### DISCUSSION TITLE
windows 环境 自定义wasm插件 编译 报错

### DISCUSSION THREAD

**Discussion by KingOfXi** [higress-group/higress#1360] [Q&A]
按照指引自定义wasm插件（https://higress.cn/docs/latest/user/wasm-go/）
1、环境
go1.19 windows/amd64
tinygo version 0.28.1 windows/amd64 

2、插件依赖执行：
go get github.com/higress-group/proxy-wasm-go-sdk
go get github.com/alibaba/higress/plugins/wasm-go@main
go get github.com/tidwall/gjson

3、编译生成wasm文件时
go mod tidy
tinygo build -o main.wasm -scheduler=none -target=wasi -gc=custom -tags="custommalloc nottinygc_finalizer" ./
但是tinygo  build  命令执行时会报 github.com/higress-group/proxy-wasm-go-sdk 该依赖的编码问题
<img width="1068" alt="编码" src="https://github.com/user-attachments/assets/cb2cfae5-3b2a-46b9-95ac-c2f8bb80e0fd">

可实际上打开该文件看到是utf-8编码。
<img width="1080" alt="utf" src="https://github.com/user-attachments/assets/7db00fb0-d760-48b5-8418-22ea92d55c02">

**Comment by CH3CHO** [comment]
建议参考一下这个 Makefile，用容器的方式构建插件，会更方便一些：https://github.com/alibaba/higress/blob/main/plugins/wasm-go/Makefile
